src/OrchardCore.Modules/OrchardCore.Users/Deployment/AllUsersDeploymentSource.cs

```diff
@@ -0,0 +1,58 @@
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Recipes;
+using YesSql;
+
+namespace OrchardCore.Users.Deployment;
+
+public class AllUsersDeploymentSource : IDeploymentSource
+{
+    private readonly ISession _session;
+
+    public AllUsersDeploymentSource(ISession session)
+    {
+        _session = session;
+    }
+
+    public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+    {
+        if (step is not AllUsersDeploymentStep allRolesStep)
+        {
+            return;
+        }
+
+        var allUsers = await _session.Query<User>().ListAsync();
+        var users = new JArray();
+
+        foreach (var user in allUsers)
+        {
+            users.Add(JObject.FromObject(
+                new UsersStepUserModel
+                {
+                    UserName = user.UserName,
+                    UserId = user.UserId,
+                    Id = user.Id,
+                    Email = user.Email,
+                    EmailConfirmed = user.EmailConfirmed,
+                    PasswordHash = user.PasswordHash,
```

Comment:
Just for info I'm working on #7891 to be able to securely exchange secrets as passwords, credentials for email, RSA keys in place of X509 certificates for OpenId and so on.

Not sure that it could be applied to user passwords, I would need to re-focus on #7891 (didn't work on it the last 2 weeks). Hmm, if it can be applied maybe in a separate PR if we agree to merge this one in the meantime.

Hmm, to deploy secrets we share a symetric key which is itself encrypted by an assymetric key (in fact one to encrypt data and onother one for signing), so maybe we could use the same technic to exchange users without having to encrypt their passwords differently.

This would mean that it is not incompatible with this PR and could be done afterwards, or maybe not, we could think that we first need #7891 and that this PR would need to use it, meaning that deploying users would need 2 RSA secrets to be defined, one for encryption and one for signing, as will be done to deploy secrets.

Reply:
Does that mean we need to postpone merging this PR?

Reply:
Not necessarily ;)

I'm not against merging this PR first, was just for info, I let others decide.